### ISSUE TITLE
How can I remove unnecessary requests when I replace a module? (Attached)

### ISSUE THREAD

**Issue by favger** [klis87/redux-requests#401]
The "resetRequests" function works for me, but the request continues to seem unnecessary. Is there a method to remove the request query from the Requests reducer?

![SS](https://user-images.githubusercontent.com/25633595/99824646-0c204780-2b67-11eb-9601-b922ad9b9da5.png)

**Comment by klis87** [comment]
You mean that state is reset, not removed from queries?

**Comment by favger** [comment]
@klis87 definitely yes

**Comment by klis87** [comment]
@favger I think this is nice to have, I will add it to TODO list. This doesn't really matter all that much though I think, memory won't be hurt for sure with this, and from selector perspective reset query and no query in reducer is the same anyway.

Is there any reason this might be important for you? Maybe I missed a case?

**Comment by klis87** [comment]
btw, this is not so simple, sometimes you could have multiple requests pending, and all aborted at once (reset also aborts requests) If you removed queries on reset, you would remove pending counter, many edge cases could be introduced with this change.

**Comment by favger** [comment]
@klis87 "I think this is nice to have, I will add it to TODO list." cool, thanks. 

"Is there any reason this might be important for you? Maybe I missed a case?": I realized it was not a problem in terms of performance. But I think it creates visual pollution in the application development process.

**Comment by klis87** [comment]
@favger I agree, definitely it is worth doing at some point. I have too many todos at the moment though, but if you wish, you could create a PR. As a hint, I think the following cases need to be covered:
1) no pending requests when reset request is fired - it is safe to remove a given query key
2) any pending request and reset requests is fired - do not remove the query key
3) in onAbort handler in query reducer, when `pending` goes from 1 to 0  when `data` is also `null`, remove query instead of just decrementing `pending`
4) do similar thing for mutations like in 3

**Comment by klis87** [comment]
@favger FYI - I am working on React branch - adding ability for `useQuery` and `useMutation` to auto dispatch request actions, together with autoclearing on unmount and on request type/request key change, plus with optional `RequestsProvider` which will remove any need to setup and use Redux directly.

This is related to this issue so I will handle this at the same time. Unless by chance you already made some work on it, then please let me know.

**Comment by favger** [comment]
@klis87 I am not working on this topic. I looked at the core code but decided there was nothing I could do. I would be very happy if you could support this issue.

**Comment by klis87** [comment]
@favger cool, so we won't duplicated each other. I will ping you here once this lands in the core!

**Comment by favger** [comment]
@klis87 We look forward to this feature. 😃

**Comment by klis87** [comment]
@favger it will probably take like 1-2 weeks, as next release will be quite huge, no breaking changes, but things like autofetching in useQuery, auto resets, suspense and suspense based SSR, possibility to hide Redux setup and treat Redux like lib's internal representation, and more!

**Comment by favger** [comment]
@klis87 Sounds great. We are impatiently awaiting new changes. 👏 👏

**Comment by klis87** [comment]
@favger see new version 1.4 for this feature

**Comment by favger** [comment]
@klis87 excellent. bless your hands.

**Comment by klis87** [comment]
@favger Did you already have a chance to test it? If yes, let's close this issue :)

**Comment by favger** [comment]
@klis87 Yes, I downloaded the new version and tested it. We can end the matter. Thank you so much.
